connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -105,32 +105,19 @@ public RestServer(WorkerConfig config) {
         createConnectors(listeners, adminListeners);
     }
 
-    @SuppressWarnings("deprecation")
-    List<String> parseListeners() {
-        List<String> listeners = config.getList(WorkerConfig.LISTENERS_CONFIG);
-        if (listeners == null || listeners.size() == 0) {
-            String hostname = config.getString(WorkerConfig.REST_HOST_NAME_CONFIG);
-
-            if (hostname == null)
-                hostname = "";
-
-            listeners = Collections.singletonList(String.format("%s://%s:%d", PROTOCOL_HTTP, hostname, config.getInt(WorkerConfig.REST_PORT_CONFIG)));
-        }
-
-        return listeners;
-    }
-
     /**
      * Adds Jetty connector for each configured listener
      */
     public void createConnectors(List<String> listeners, List<String> adminListeners) {
         List<Connector> connectors = new ArrayList<>();
 
-        for (String listener : listeners) {
-            if (!listener.isEmpty()) {
-                Connector connector = createConnector(listener);
-                connectors.add(connector);
-                log.info("Added connector for {}", listener);
+        if (listeners != null && !listeners.isEmpty()) {
+            for (String listener : listeners) {
+                if (!listener.isEmpty()) {
+                    Connector connector = createConnector(listener);
+                    connectors.add(connector);
+                    log.info("Added connector for {}", listener);
+                }
```

Comment:
Are we just relying upon worker config validation to ensure that at least one listener is required? If so, then do we need the outer condition? If not, then should we add those checks via validators so configuration errors are checked immediately upon startup?

Reply:
Added the tests and validator